mbta/lines.go

```diff
@@ -0,0 +1,110 @@
+package mbta
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const linesAPIPath = "/lines"
+
+// LineService handling all of the route related API calls
+type LineService service
+
+// Line holds all the info about a given MBTA Route
+type Line struct {
+	ID        string  `jsonapi:"primary,line"`
+	Color     string  `jsonapi:"attr,color"`
+	LongName  string  `jsonapi:"attr,long_name"`
+	ShortName string  `jsonapi:"attr,short_name"`
+	SortOrder int     `jsonapi:"attr,sort_order"`
+	TextColor string  `jsonapi:"attr,text_color"`
+	Routes    []Route `jsonapi:"relation,route`
+}
+
+type LineInclude string
+
+const (
+	LineIncludeRoutes LineInclude = includeRoutes
+)
+
+// GetAllLinesSortByType all of the possible ways to sort by for a GetAllLines request
+type GetAllLinesSortByType string
+
+const (
+	GetAllLinesSortByColorAscending      GetAllLinesSortByType = "color"
+	GetAllLinesSortByColorDescending     GetAllLinesSortByType = "-color"
+	GetAllLinesSortByLongNameDesending   GetAllLinesSortByType = "long_name"
+	GetAllLinesSortByLongNameDescending  GetAllLinesSortByType = "-long_name"
+	GetAllLinesSortByShortNameAscending  GetAllLinesSortByType = "short_name"
+	GetAllLinesSortByShortNameDescending GetAllLinesSortByType = "-short_name"
+	GetAllLinesSortBySortOrderAscending  GetAllLinesSortByType = "sort_order"
+	GetAllLinesSortBySortOrderDescending GetAllLinesSortByType = "-sort_order"
+	GetAllLinesSortByTextColorAscending  GetAllLinesSortByType = "text_color"
+	GetAllLinesSortByTextColorDescending GetAllLinesSortByType = "-text_color"
+)
+
+// GetAllLinesRequestConfig extra options for the GetAllLines request
+type GetAllLinesRequestConfig struct {
+	PageOffset string                `url:"page[offset],omitempty"`       // Offset (0-based) of first element in the page// Offset (0-based) of first element in the page
+	PageLimit  string                `url:"page[limit],omitempty"`        // Max number of elements to return// Max number of elements to return
+	Sort       GetAllLinesSortByType `url:"sort,omitempty"`               // Results can be sorted by the id or any GetAllRoutesSortByType
+	Fields     []string              `url:"fields[line],comma,omitempty"` // Fields to include with the response. Note that fields can also be selected for included data types// Fields to include with the response. Multiple fields MUST be a comma-separated (U+002C COMMA, “,”) list. Note that fields can also be selected for included data types
+	Include    []LineInclude         `url:"include,comma,omitempty"`      // Include extra data in response
+	FilterIDs  []string              `url:"filter[id],comma,omitempty"`   // Filter by multiple IDs
+}
+
+// GetLineRequestConfig extra options for the GetAllLines request
+type GetLineRequestConfig struct {
```

Comment:
Move this down to line 88ish so it is above the GetLine functions. Just to stick with the mindset of keeping things as close to where they are used as possible